lib/ansible/modules/cloud/docker/docker_container.py

```diff
@@ -2089,10 +2095,12 @@ def has_different_configuration(self, image):
 
         host_config = self.container['HostConfig']
         log_config = host_config.get('LogConfig', dict())
-        restart_policy = host_config.get('RestartPolicy', dict())
         config = self.container['Config']
         network = self.container['NetworkSettings']
 
+        if self.parameters.client.docker_api_version < LooseVersion('1.22'):
+            restart_policy = host_config.get('RestartPolicy', dict())
```

Comment:
Please also move this into the `if` below which updates `config_mapping`.

Reply:
Done. Many thanks for the review and all your input! Some of them are smart solutions to things I was not completely satisfied by before!